File: ServiceCore/ProvinceService.cs
using FacadeCore;
using BOCore;
using DALCore;

namespace ServiceCore
{
    public class ProvinceService : IProvinceService
    {
        private readonly UnitOfWorkCore _uow;

        public ProvinceService(UnitOfWorkCore uow)
        {
            _uow = uow;
        }

        public GetResponse<IdNameBO> GetProvinces()
        {
            var response = new GetResponse<IdNameBO>();

            // No tracking, gewoon alles naar IdNameBO mappen
            var entities = _uow.Provinces.GetNoTracking()
                .OrderBy(p => p.ProvincieName);
            foreach (var e in entities)
                response.AddValue(e.GetIdName());

            return response;
        }
    }
}

File: Service/ProvinceService.vb
Imports DAL
Imports BO
Imports Facade

Public Class ProvinceService
    Implements IProvinceService


    Public Function GetProvinces() As GetResponse(Of IdNameBO) Implements IProvinceService.GetProvinces
        Dim response As New GetResponse(Of IdNameBO)
        Dim uow As New UnitOfWork(False)
        Dim dao = uow.GetProvinceDAO()
        Dim entities = dao.GetNoTracking()
        For Each _entity In entities
            response.AddValue(_entity.GetIdName())
        Next
        Return response
    End Function
End Class
